File: core/src/main/java/com/batchsphere/core/batch/entity/Batch.java
package com.batchsphere.core.batch.entity;

import com.batchsphere.core.masterdata.material.entity.Material;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name="batch")
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Batch {

    @Id
    @Column(nullable = false, updatable = false)
    private UUID id;

    @Column(name = "batch_number", nullable = false, unique = true, length = 100)
    private String batchNumber;

    @ManyToOne(fetch= FetchType.LAZY)
    @JoinColumn(name = "material_id", nullable = false)
    private Material material;

    @Enumerated(EnumType.STRING)
    @Column(name = "batch_type", nullable = false)
    private BatchType batchType;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private BatchStatus batchStatus;

    @Column(name = "quantity", nullable = false, precision = 18, scale = 3)
    private BigDecimal quantity;

    @Column(name = "unit_of_measure", nullable = false, length = 20)
    private String unitOfMeasure;

    @Column(name = "manufacture_date")
    private LocalDate manufactureDate;

    @Column(name = "expiry_date")
    private LocalDate expiryDate;

    @Column(name = "retest_date")
    private LocalDate retestDate;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    @Column(name = "created_by", nullable = false)
    private String createdBy;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;



}

File: core/ui/src/types/batch.ts
export type BatchType = "RAW_MATERIAL" | "FINISHED_GOOD";

export type Batch = {
  id: string;
  batchNumber: string;
  material: {
    id: string;
    materialCode: string;
    materialName: string;
  } | null;
  batchType: BatchType;
  batchStatus: string;
  quantity: number;
  unitOfMeasure: string;
  manufactureDate: string | null;
  expiryDate: string | null;
  retestDate: string | null;
  isActive: boolean;
  createdBy: string;
  createdAt: string;
  updatedBy: string | null;
  updatedAt: string | null;
};

export type CreateBatchRequest = {
  batchNumber: string;
  materialId: string;
  batchType: BatchType;
  quantity: number;
  unitOfMeasure: string;
  manufactureDate?: string;
  expiryDate?: string;
  retestDate?: string;
  createdBy: string;
};
